lib/components/user/user-account-screen.js

```diff
@@ -0,0 +1,437 @@
+      if (result.status === 'success') {
+        alert('Your preferences have been saved.')
+      } else {
+        alert(`An error was encountered:\n${result.message}`)
+      }
+    }
+  }
+
+  _handleConsentChange = e => {
+    this._updateUserState({ hasConsentedToTerms: e.target.checked })
+  }
+
+  _handleExit = () => {
+    const { originalUrl } = this.props
+    this.props.routeTo(originalUrl)
+  }
+
+  _handleExitAndSave = async () => {
+    await this._updateUserPrefs()
+    this._handleExit()
+  }
+
+  _handleLocationsChange = newLocations => {
+    this._updateUserState({ savedLocations: newLocations })
+  }
+
+  _handleNotificationChannelChange = e => {
+    this._updateUserState({ notificationChannel: e })
+  }
+
+  _handleToNextPane = panes => {
+    return async () => {
+      const { activePaneId } = this.state
+      const nextId = panes[activePaneId].nextId
+
+      if (nextId) {
+        this.setState({
+          activePaneId: nextId
+        })
+      } else {
+        await this._handleExitAndSave()
+      }
+    }
+  }
+
+  _handlePhoneNumberChange = e => {
+    this._updateUserState({ phoneNumber: e.target.value })
+  }
+
+  _handleStoreTripsChange = e => {
+    this._updateUserState({ storeTripHistory: e.target.checked })
+  }
+
+  _handleToPrevPane = panes => {
+    return () => {
+      const { activePaneId } = this.state
+      this.setState({
+        activePaneId: panes[activePaneId].prevId
+      })
+    }
+  }
+
+  _makePanes = () => {
+    const {
+      isNewAccount,
+      userData
+    } = this.state
+
+    const {
+      hasConsentedToTerms,
+      notificationChannel = 'email',
+      phoneNumber,
+      savedLocations,
+      storeTripHistory
+    } = userData
+
+    const terms = (
+      <TermsOfUsePane
+        checkHistory={storeTripHistory}
+        checkTerms={hasConsentedToTerms}
+        disableCheckTerms={!isNewAccount}
+        onCheckHistoryChange={this._handleStoreTripsChange}
+        onCheckTermsChange={this._handleConsentChange}
+      />
+    )
+
+    const notifications = (
+      <NotificationPrefsPane
+        email={userData.email}
+        notificationChannel={notificationChannel}
+        onNotificationChannelChange={this._handleNotificationChannelChange}
+        onPhoneNumberChange={this._handlePhoneNumberChange}
+        phoneNumber={phoneNumber}
+      />
+    )
+
+    const verifyPhone = <PhoneVerificationPane />
+
+    const locations = (
+      <FavoriteLocationsPane
+        locations={savedLocations}
+        onLocationsChange={this._handleLocationsChange}
+      />
+    )
+
+    const finish = <AccountSetupFinishPane />
+
+    return {
+      terms,
+      notifications,
+      verifyPhone,
+      locations,
+      finish
+    }
+  }
+
+  _renderNewAccount = () => {
+    const {
+      activePaneId,
+      userData
+    } = this.state
+
+    const {
+      hasConsentedToTerms,
+      notificationChannel = 'email'
+    } = userData
+
+    const panes = this._makePanes()
```

Comment:
This will go through the process of generating the react for every pane of every render. See comments in how to change `_makePanes()`.